slasher/db/kv/spanner.go

```diff
@@ -0,0 +1,282 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/boltdb/bolt"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/slasher/detection/attestations/types"
+	log "github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+)
+
+var highestEpoch uint64
+
+func saveToDB(db *Store) func(uint64, uint64, interface{}, int64) {
+	// Returning the function here so we can access the DB properly from the OnEvict.
+	return func(epoch uint64, _ uint64, value interface{}, cost int64) {
+		log.Tracef("evicting span map for epoch: %d", epoch)
+		err := db.update(func(tx *bolt.Tx) error {
+			bucket := tx.Bucket(validatorsMinMaxSpanBucket)
+			epochBucket, err := bucket.CreateBucketIfNotExists(bytesutil.Bytes8(epoch))
+			if err != nil {
+				return err
+			}
+			spanMap := value.(map[uint64]types.Span)
+			for k, v := range spanMap {
+				err = epochBucket.Put(bytesutil.Bytes8(k), marshalSpan(v))
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+		if err != nil {
+			log.Errorf("failed to save span map to db on cache eviction: %v", err)
+		}
+	}
+}
+
+func unmarshalSpan(ctx context.Context, enc []byte) (types.Span, error) {
+	ctx, span := trace.StartSpan(ctx, "SlasherDB.unmarshalSpan")
+	defer span.End()
+	r := types.Span{}
+	if len(enc) != 7 {
```

Comment:
why 7? what is the meaning of this number? Can it be a constant or something instead of a magic number?

Reply:
added a const in schema. thanks